rust/src/treefile.rs

```diff
@@ -1226,6 +1259,66 @@ etc-group-members:
                 == "table"
         );
     }
+
+    #[test]
+    fn test_split_whitespace_unless_quoted() {
+        // test single quoted package
+        let single_quoted_pkg = String::from("'foobar >= 1.0'");
+        let pkgs: Vec<String> = split_whitespace_unless_quoted(&single_quoted_pkg).collect();
```

Comment:
Also it would make sense for `split_whitespace_unless_quoted` to take an `&str` and not a `&String` - that way the test cases like this can just pass a static pointer rather than an allocated string.